server.go

```diff
@@ -962,7 +962,7 @@ func (s *Server) processSubscribe(cl *Client, pk packets.Packet) error {
 		code = packets.ErrPacketIdentifierInUse
 	}
 
-	existed := false
+	existedMarks := make([]bool, len(pk.Filters))
```

Comment:
I'm going to change this var name to `filterExisted` for clarity before release, but otherwise this is all a great improvement! 👍🏻 Thank you very much @wind-c!